File: app/src/main/java/ch/unstable/ost/about/AboutFragment.java
package ch.unstable.ost.about;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import ch.unstable.ost.BuildConfig;
import ch.unstable.ost.R;

/**
 * A placeholder fragment containing a simple view.
 */
public class AboutFragment extends Fragment {

    public AboutFragment() {
    }

    /**
     * Returns a new instance of this fragment for the given section
     * number.
     */
    public static AboutFragment newInstance() {
        return new AboutFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_about, container, false);
        TextView version = rootView.findViewById(R.id.app_version);
        version.setText(BuildConfig.VERSION_NAME);

        View githubLink = rootView.findViewById(R.id.github_link);
        githubLink.setOnClickListener(new OnGithubLinkClickListener());

        View licenseLink = rootView.findViewById(R.id.app_read_license);
        licenseLink.setOnClickListener(new OnReadFullLicenseClickListener());
        return rootView;
    }

    private static class OnGithubLinkClickListener implements View.OnClickListener {
        @Override
        public void onClick(final View view) {
            final Context context = view.getContext();
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(context.getString(R.string.github_url)));
            context.startActivity(intent);
        }
    }

    private static class OnReadFullLicenseClickListener implements View.OnClickListener {
        @Override
        public void onClick(View v) {
            LicenseFragment.Companion.showLicense(v.getContext(), StandardLicenses.GPL3);
        }
    }
}

File: app/src/main/java/ch/unstable/ost/about/AboutFragment.kt
package ch.unstable.ost.about

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import ch.unstable.ost.BuildConfig

import ch.unstable.ost.R
import kotlinx.android.synthetic.main.fragment_about.*

/**
 * A placeholder fragment containing a simple view.
 */
class AboutFragment : Fragment() {

    private val appLicense = StandardLicenses.GPL3;

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_about, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        app_version.text = BuildConfig.VERSION_NAME

        github_link.setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(it.context.getString(R.string.github_url)))
            context!!.startActivity(intent)
        }

        app_read_license.setOnClickListener {
            LicenseFragment.showLicense(it.context, appLicense)
        }
    }

    companion object {

        /**
         * Returns a new instance of this fragment for the given section
         * number.
         */
        fun newInstance(): AboutFragment {
            return AboutFragment()
        }
    }
}
